zetaclient/bitcoin_client.go

```
@@ -379,17 +382,19 @@ func (ob *BitcoinChainClient) IsSendOutTxProcessed(sendHash string, nonce uint64
 	}
 
 	var amount float64
-	if res.Amount > 0 {
-		ob.logger.ObserveOutTx.Warn().Msg("IsSendOutTxProcessed: res.Amount > 0")
+	if res.Amount >= 0 {
+		ob.logger.ObserveOutTx.Warn().Msgf("IsSendOutTxProcessed: res.Amount >= 0")
 		amount = res.Amount
-	} else if res.Amount == 0 {
-		ob.logger.ObserveOutTx.Error().Msg("IsSendOutTxProcessed: res.Amount == 0")
-		return false, false, nil
 	} else {
 		amount = -res.Amount
 	}
 
-	amountInSat, _ := big.NewFloat(amount * 1e8).Int(nil)
+	sats, err := getSatoshis(amount)
+	if err != nil {
+		ob.logger.ObserveOutTx.Warn().Msgf("IsSendOutTxProcessed: getSatoshis error: %s", err)
+		return false, false, nil
```

Shouldn't we return an error here?

---

sure